presto-hive-metastore/src/main/java/com/facebook/presto/hive/MetastoreClientConfig.java

```diff
@@ -317,4 +324,60 @@ public MetastoreClientConfig setInvalidateMetastoreCacheProcedureEnabled(boolean
         this.invalidateMetastoreCacheProcedureEnabled = invalidateMetastoreCacheProcedureEnabled;
         return this;
     }
+
+    @Config("hive.metastore.thrift.client.tls.enabled")
+    @ConfigDescription("Enable TLS security for HMS")
+    public MetastoreClientConfig setTlsEnabled(boolean tlsEnabled)
+    {
+        this.tlsEnabled = tlsEnabled;
+        return this;
+    }
+    public boolean isTlsEnabled()
+    {
+        return tlsEnabled;
+    }
+    @Config("hive.metastore.thrift.client.tls.keystore.path")
+    @ConfigDescription("Path to JKS or PEM key store")
+    public MetastoreClientConfig setKeystorePath(File keystorePath)
+    {
+        this.keystorePath = keystorePath;
+        return this;
+    }
+    public File getKeystorePath()
+    {
+        return keystorePath;
+    }
+    @Config("hive.metastore.thrift.client.tls.keystore.password")
+    @ConfigDescription("Password to key store")
+    public MetastoreClientConfig setKeystorePassword(String keystorePassword)
+    {
+        this.keystorePassword = keystorePassword;
+        return this;
+    }
+    public String getKeystorePassword()
+    {
+        return keystorePassword;
+    }
+    @Config("hive.metastore.thrift.client.tls.truststore.path")
+    @ConfigDescription("Path to JKS or PEM trust store")
+    public MetastoreClientConfig setTruststorePath(File truststorePath)
+    {
+        this.truststorePath = truststorePath;
+        return this;
+    }
+    public File getTruststorePath()
+    {
+        return truststorePath;
+    }
+    @Config("hive.metastore.thrift.client.tls.truststore.password")
+    @ConfigDescription("Path to trust store")
+    public MetastoreClientConfig setTrustStorePassword(String truststorePassword)
+    {
+        this.trustStorePassword = truststorePassword;
+        return this;
+    }
+    public String getTrustStorePassword()
+    {
+        return trustStorePassword;
+    }
```

Comment:
This comment is for line 302 to 314
Move "hive.metastore.thrift.delete-files-on-table-drop" to the new com/facebook/presto/hive/metastore/thrift/ThriftHiveMetastoreConfig.java

I'm not sure if this is only for the thrift client? If it is then it's better to rename it to "hive.metastore.thrift.client.delete-files-on-table-drop" and deprecate "hive.metastore.thrift.delete-files-on-table-drop". I think it is not. @imjalpreet Could you please confirm?

Reply:
Yes, we can move this to the `ThriftHiveMetastoreConfig`. The config is only valid when Thrift Hive Metastore is used as the metastore in Hive connector due to a limitation observed a few years back: https://github.com/prestodb/presto/pull/17369